src/Command/ModuleDeployAfterMagentoCommand.php

```diff
@@ -43,83 +112,152 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $startTime = microtime(true);
 
-        $projectRoot = getcwd();
-        $directoryName = basename($projectRoot);
+        $mainDomain = basename(getcwd());
+        $modulesFolder = $input->getArgument(self::OPTION_FOLDER);
+
+        $together = $input->getOption(self::OPTION_TOGETHER);
 
-        # Stage 1: copy files, deploy Sample Data, create database
-        if (!file_exists('./app') || !is_dir('./app')) {
-            $output->writeln('<error>Seems we\'re not inside the Magento project</error>');
-            exit;
+        if (file_exists('./composer.json')) {
+            $magentoComposerFile = json_decode(file_get_contents('./composer.json'));
+
+            if (!is_dir('./app') || strpos($magentoComposerFile->name, 'magento/') === false) {
+                $output->writeln('<error>Seems we\'re not inside the Magento project</error>');
+                exit(1);
+            }
+        } else {
+            $output->writeln('<error>Composer.json not found</error>');
+            exit(1);
         }
 
-        // @TODO: use composer require to install from the repository
-        // @TODO: add directory parameter with all modules
-        // @TODO: find all etc/module.xml, get vendor-module, create folders, copy
+        if (!is_dir($modulesFolder)) {
+            $output->writeln('<error>Modules directory does exist.</error>');
+            exit(1);
+        }
 
-        $output->writeln('<info>Cleanup Magento 2 application...</info>');
+        $modules = $this->detectModules($modulesFolder);
 
-        $this->dockerExec("sh -c \"rm -rf dirlisthere\"");
-        $directoriesToClean = [
+        if (empty($modules)) {
+            $output->writeln('<error>Modules directory is empty.</error>');
+            exit(1);
+        }
 
-        ];
+        $this->ask(MysqlContainer::QUESTION, $input, $output);
 
-        // @TODO: additional directories to remove
-        // @TODO: do we need additional external packages to require? Or probably must get them from the composer.json of the installed modules???????????
+        # Stage 0: clean Magento 2, install Magento application, handle together attribute
+        $output->writeln('<info>Cleanup Magento 2 application...</info>');
 
-        passthru(<<<BASH
-            rm -rf ./app/code/Vendor
-            rm -rf ./pub/media/vendor/
-            mkdir -p ./app/code/Vendor
-BASH
-        );
+        $filesAndFolderToRemove = implode(' ', $this->magentoDirectoriesAndFilesToClean);
+        $this->shell->dockerExec("sh -c \"rm -rf {$filesAndFolderToRemove}\"", $mainDomain);
+
+        if ($together) {
+            $output->writeln('<info>Copying modules to run installation together with the Magento...</info>');
+            $this->copyModules($modules, $mainDomain);
+        }
 
+        $output->writeln('<info>Reinstalling Magento 2 application...</info>');
```

Comment:
@serhiira Incorrect execution order: modules are copied BEFORE this line, but Sample Data is installed only AFTER that.